src/main/java/io/cdap/plugin/gcp/gcs/sink/DelegatingGCSOutputCommitter.java

```diff
@@ -0,0 +1,108 @@
+import org.apache.hadoop.mapreduce.TaskAttemptContext;
+import org.apache.hadoop.mapreduce.lib.output.FileOutputFormat;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Output Committer which creates and delegates operations to other GCS Output Committer instances.
+ *
+ * Delegated instances are created based on a supplied Output Format and Destination Table Names.
+ */
+public class DelegatingGCSOutputCommitter extends OutputCommitter {
+  private final Map<String, OutputCommitter> committerMap;
+
+  public DelegatingGCSOutputCommitter() {
+    committerMap = new HashMap<>();
+  }
+
+  /**
+   * Add a new GCSOutputCommitter based on a supplied Input Format and Table Name.
+   *
+   * This GCS Output Committer gets initialized when created.
+   */
+  @SuppressWarnings("rawtypes")
+  public void addGCSOutputCommitterFromOutputFormat(OutputFormat outputFormat,
+                                                    TaskAttemptContext context,
+                                                    String tableName) throws IOException, InterruptedException {
+    //Set output directory
+    context.getConfiguration().set(FileOutputFormat.OUTDIR,
+                                   DelegatingGCSOutputUtils.buildOutputPath(context.getConfiguration(), tableName));
+
+    //Wrap output committer into the GCS Output Committer.
+    GCSOutputCommitter gcsOutputCommitter = new GCSOutputCommitter(outputFormat.getOutputCommitter(context));
+
+    //Initialize the new GCS Output Committer and add it to the Committer Map
+    gcsOutputCommitter.setupJob(context);
+    gcsOutputCommitter.setupTask(context);
+    committerMap.put(tableName, gcsOutputCommitter);
+  }
+
+  @Override
+  public void setupJob(JobContext jobContext) throws IOException {
+    //no-op
+  }
+
+  @Override
+  public void setupTask(TaskAttemptContext taskAttemptContext) throws IOException {
+    //no-op
+  }
+
+  @Override
+  public boolean needsTaskCommit(TaskAttemptContext taskAttemptContext) throws IOException {
+    if (committerMap.isEmpty()) {
+      return false;
+    }
+
+    boolean needsTaskCommit = true;
+
+    for (OutputCommitter committer : committerMap.values()) {
+      needsTaskCommit = needsTaskCommit && committer.needsTaskCommit(taskAttemptContext);
+    }
+
+    return needsTaskCommit;
+  }
+
+  @Override
+  public void commitTask(TaskAttemptContext taskAttemptContext) throws IOException {
+    for (OutputCommitter committer : committerMap.values()) {
+      committer.commitTask(taskAttemptContext);
+    }
+  }
+
+  @Override
+  public void commitJob(JobContext jobContext) throws IOException {
+    for (OutputCommitter committer : committerMap.values()) {
+      committer.commitJob(jobContext);
+    }
+  }
+
+  @Override
+  public void abortTask(TaskAttemptContext taskAttemptContext) throws IOException {
+    for (OutputCommitter committer : committerMap.values()) {
+      committer.abortTask(taskAttemptContext);
```

Comment:
if one abort throws an exception, we still want to try to abort the others.

Reply:
Fixed.